packages/components/src/internal/app/utils.ts

```diff
@@ -124,6 +124,10 @@ export function isSampleManagerEnabled(): boolean {
     return getServerContext().moduleContext?.samplemanagement !== undefined;
 }
 
+export function hasPremiumModule(): boolean {
```

Comment:
nit: I would consider making this into a reusable hook.

```ts
export function hasPremiumModule(): boolean {
    const { moduleContext } = getServerContext();
    return useMemo(() => moduleContext?.samplemanagement?.hasPremiumModule ?? false, [moduleContext]);
}
```

Reply:
Nice. Done.